src/protocol.h

```diff
@@ -120,6 +120,11 @@ extern const char *GETHEADERS;
  * @see https://bitcoin.org/en/developer-reference#tx
  */
 extern const char *TX;
+/**
+ * The dandelion tx message transmits a single transaction.
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0156.mediawiki
+ */
+extern const char* TX_DANDELION;
```

Comment:
I think in the past, we've usually bumped the protocol version when introducing new p2p messages so that we only send the new messages (like ACCEPT_DANDELION) to peers that are signaling that they've upgraded.  Not sure how important that is...

Reply:
Note that we'd never send a `TX_DANDELION` to a peer that never sent us `ACCEPT_DANDELION` in the first place. So I don't think this is an issue at all.

Reply:
Perhaps my comment was not placed on the right line of this file; I was thinking of the dandelionacc message.  With sendheaders and I think compact blocks, I believe we bumped the protocol version and only tried to send the new p2p messages (even for handshaking/negotiating the protocol to use) to peers with that higher version?

Reply:
Good point. I think it is a lot easier to do this without enforcing a global network version bump.

Note that the `dandelionacc` message is properly read by all nodes that implement dandelion (regardless of their network protocol version) or choose to ignore that message. And for all other nodes (that are completely unaware of dandelion) they simply treat this as unknown message (once per connection) in their logs. I think this one debug log is not enough reason to go through the hassle of bumping the protocol version.

https://github.com/bitcoin/bitcoin/blob/63f8b0128b2aac3b25c6ec4d2f5bda213033162a/src/net_processing.cpp#L2942